cmd/telemetrygen/internal/common/log.go

```diff
@@ -4,20 +4,26 @@
 package common
 
 import (
-	"fmt"
-
 	grpcZap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
 	"go.uber.org/zap"
 )
 
-// CreateLogger creates a logger for use by telemetrygen
```

Comment:
Are the logger changes necessary for this PR? We generally have avoided using `init` in the Collector/telemetrygen codebases, reasoning about package initialization order is somewhat complex

Reply:
We had a data race because the logger when created registers as a singleton for gRPC logging. The tests I added trigger enough parallelization that the build failed because of it.

Reply:
Would be great to see the data race, but there is no need to use `init` to fix this AFAICT. It seems likely that changing where/when we do https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/cd25ada2c983be0616c824ba547c01e11cdb3196/cmd/telemetrygen/internal/common/log.go#L19-L21 should be enough to fix this.